test/extended/authorization/rbac/groups_default_rules.go

```diff
@@ -182,7 +183,18 @@ var _ = g.Describe("[sig-auth][Feature:OpenShiftAuthorization] The default clust
 
 	oc := exutil.NewCLI("default-rbac-policy")
 
-	g.It("should have correct RBAC rules", func() {
+	g.It(fmt.Sprintf("should have correct RBAC rules [apigroup:%s][apigroup:%s][apigroup:%s][apigroup:%s][apigroup:%s][apigroup:%s][apigroup:%s][apigroup:%s][apigroup:%s][apigroup:%s]",
```

Comment:
This is quite complex. `allUnauthenticatedRules` is the source of so many groups. I'd rather turn `allUnauthenticatedRules` into a api-group-key map and loop over all apigroups over `g.It("should have correct RBAC rules", func() {` rather than have all groups in a single test name.

Reply:
I don't think it's safe to break the test apart. I am not sure why these groups are listed here though, because you don't actually need the API groups, you only need the RBAC group, the rest are just strings.

Reply:
You are right. Also, I don't see an easy way how to run the test minus missing apigroups since the test is doing set diff of both "what's currently available" and "what's expected to be available".

Reply:
Obvious approach is to dynamically detect the apigroups based on `exutil.DoesApiResourceExist`, constructing both `allAuthenticatedRules` and `allUnauthenticatedRules` based on the apigroup presence and then running the comparision/coverage. Though, there's more than just api group detection which needs to be taken into account. E.g. https://github.com/openshift/origin/blob/master/test/extended/authorization/rbac/groups_default_rules.go#L91-L94.

https://github.com/openshift/origin/blob/master/test/extended/authorization/rbac/groups_default_rules.go#L97-L100 can be removed. Maybe the legacy groups in https://github.com/openshift/origin/blob/master/test/extended/authorization/rbac/groups_default_rules.go#L59-L66 as well? What about https://github.com/openshift/origin/blob/master/test/extended/authorization/rbac/groups_default_rules.go#L134-L142?

Reply:
I'm taking a second look at this. It look like @stlaz is right, all [apigroup:*] seems wrong. Currently it complains about test data being too big which a nice mechanism except not right against microshift

Reply:
Reverted these changes as this wasn't exercising [apigroups] - it was exercising RBAC about those apigroups so the test should run on MicroShift. If it fails then we should  take a look what should we fix in MicroShift and how should we refactor test if needed